homeassistant/core.py

```diff
@@ -1257,6 +1265,65 @@ def as_dict(self) -> Dict:
             'config_source': self.config_source
         }
 
+    def _update(self, values: Dict, source: str) -> None:
+        """Update the configuration from a dictionary.
+
+        Async friendly.
+        """
+        from homeassistant.config import _set_time_zone
+        self.config_source = source
+        self.latitude = values['latitude']
+        self.longitude = values['longitude']
+        self.elevation = values['elevation']
+        unit_system = values['unit_system']
+        if unit_system == CONF_UNIT_SYSTEM_IMPERIAL:
+            self.units = IMPERIAL_SYSTEM
+        else:
+            self.units = METRIC_SYSTEM
+        self.location_name = values['location_name']
+        _set_time_zone(self.hass, values['time_zone'])
+
+    async def update(self, values: Dict) -> None:
+        """Update the configuration from a dictionary.
+
+        Async friendly.
+        """
+        from homeassistant.config import SOURCE_STORAGE
+        self._update(values, SOURCE_STORAGE)
+        self.store()
+        self.hass.bus.async_fire(
+            EVENT_CORE_CONFIG_UPDATE,
+            values
+        )
+
+    async def load(self) -> None:
+        """Load [homeassistant] core config."""
+        from homeassistant.config import SOURCE_STORAGE
+        store = self.hass.helpers.storage.Store(
+            CORE_STORAGE_VERSION, CORE_STORAGE_KEY, private=True)
+        data = await store.async_load()
+        if not data:
+            return
+
+        self._update(data, SOURCE_STORAGE)
+
+    async def store(self) -> None:
+        """Store [homeassistant] core config."""
+        config = self.as_dict()
```

Comment:
Why do this if you're a method of the object.

Reply:
Fixed.